libraries/AP_HAL_ChibiOS/hwdef/JHEM_JHEF405/readme.md

```diff
@@ -0,0 +1,111 @@
+# JHEMCU GF16-BMI stack F405 (JHEM_JHEF405) Flight Controller with 13A ESC
+
+JHEMCU GF16-BMI stack F405 Flight control 13A ESC
+produced by  [JHEMCU](https://www.jhemcu.com/)
+ 
+There are 5 serial ports (UART1, UART2, UART3, UART4, UART6), 8 motor ports, 1 I2C port, and 1 SBUS interface (the serial port used by SBUS is RX1), with WS2812 interface reserved. Built in black box 16MB and barometer, with buzzer interface and camera interface.
+
+
+![JHEMCU GF16-BMI stack](jhemcu-gf16-bmi.jpg "JHEMCU GF16-BMI stack")
+
+### Specs:
+**Flight control parameters:**
+* MCU: STM32F405
+* Gyroscope/Accelerometer: BMI270
+* OSD: AT7456E
+* Barometer: BMP280
+* Black box: 16MB
+* I2C: Support
+* BEC: 5V/3A
+* UART: UART1 (RC), UART2, UART3, UART4, UART6
+* USB: micro USB
+* Size: 16 * 16MM M3
+* Receiver: ELRS (CRSF), TBS (CRSF), SBUS, IBUS, DSM2, DSMX
+* Support programmable LED such as WS2812
+* Support buzzer
+* Built-in voltage and current sensors
+
+**Electrical adjustment parameters:**
+ 
+* Input voltage: 2-4S LiPo/2-4S HV LiPo
+* BEC output: None
+* Continuous working current of single electric regulator: 13A * 4
+* External electrical adjustment size: 24 * 25mm (including the maximum size of solder pads and protruding points)
+* Screw hole pitch 16 * 16MM, hole diameter 3MM
+* Firmware: BLHELI_ S G_ H_ 30_ 16_ 7. HEX
+* Supports Dshot600, Dshot300, Dshot150, Oneshot125, Multispot, PWM control signals
+ 
+Weight: flight control+electric adjustment=5.8g
+
+
+## Pinout
+
+![JHEMCU GF16-BMI stack](jhemcu-gf16-bmi-pinout.jpg "JHEMCU GF16-BMI stack Pinout")
+![JHEMCU GF16-BMI stack](jhemcu-gf16-bmi-8axis.jpg "JHEMCU GF16-BMI stack 8-axis frame")
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+|Name|Pin|Function|
+|:-|:-|:-|
+|SERIAL0|COMPUTER|USB|
+|SERIAL1|RX1/TX1|USART1 (DMA) - RC Input/Output |
+|SERIAL2|TX2/RX2|USART2 (DMA) - User |
+|SERIAL3|TX3/RX3|USART3 (DMA) - User |
+|SERIAL4|TX4/RX4|UART4 (DMA) - User |
```

Comment:
```suggestion
|SERIAL4|TX4/RX4|UART4 (DMA TX only) - GPS2 |
```

Reply:
and you did not adddress my previous comment...based on your wiring diagram you expect users to use this port for VTX control...you should set the protocol in the hwdef to DEFAULT_SERIAL4_PROTOCOL SerialProtocol_Tramp or _SmartAudio

Reply:
can we keep the port just as a general purpose port, so users will decide how they will use it?

Reply:
you can but BF users using your wiring diagram will have a harder time transitioning to AP